### DISCUSSION TITLE
Is it possible to write feature selection logs to file?

### DISCUSSION THREAD

**Discussion by AlexGrunt** [rasbt/mlxtend#1048] [Q&A] [answered]
Hello!

I want to know if is it possible to send `SequentialFeatureSelector.fit()` method logs to specified logfile. 

I'm using this method in Jupyter notebook and I can not longer see the output of this method after Reconnect to Jupyter Kernel. Moreover if I trying to interrupt current run and call `SequentialFeatureSelector.finalize_fit()`, `SequentialFeatureSelector.get_metric_dict()` method isn't accessible for it's object and I can't get feature selection history for this run. 

Is there any workaround for described usecase?

Any help is greatly appreciated!

**Comment by arilwan** [comment, marked as answer]
HI @AlexGrunt,

Yes, I have done this and it works fine. Here's how:

1.  create a file handler object (e.g. `f = open(...)`  just before the `try...except`  block around [here](https://github.com/rasbt/mlxtend/blob/master/mlxtend/feature_selection/sequential_feature_selector.py#L534).
2. Replace the `sys.stderr.write` (or add) with `f.write(...)` around [line](https://github.com/rasbt/mlxtend/blob/master/mlxtend/feature_selection/sequential_feature_selector.py#L625).   Other option could be to redirect the `stderr` out in your terminal to a file.
In the first case, do not forget to write to the file every time there update (don't wait until the is completed), e.g.
````
f.write(f'\n{k_idx} \t{cv_scores} \t{k_score}')
f.flush()
os.fsync(f.fileno()) # last 2 lines to ensure history is written to the file
````

Hope this helps.

**Reply by AlexGrunt** [reply]
Thank you
